### PR TITLE
`IITOFFOLI` gate

### PR DESCRIPTION
**Pull request by vtomole**
Fixes: https://github.com/SupertechLabs/cirq-superstaq/issues/206

### CODE REVIEW

**Comment by singular-value** [comment]
@vtomole is this ready? (well I guess we'll want qss one too for parity)

**Comment by vtomole** [comment]
@singular-value 

This will be done once `cirq.testing.assert_has_consistent_apply_unitary(gate)` and `cirq.testing.assert_decompose_is_consistent_with_unitary(gate, ignoring_global_phase=True)` pass; which they don't at the moment. Will need to revist my definition of this gate's unitary. Right now i define it as
```
[[0 i 0 0 0 0 0 0]
 [i 0 0 0 0 0 0 0]
 [0 0 1 0 0 0 0 0]
 [0 0 0 1 0 0 0 0]
 [0 0 0 0 1 0 0 0]
 [0 0 0 0 0 1 0 0]
 [0 0 0 0 0 0 1 0]
 [0 0 0 0 0 0 0 1]]
``` 
but the correct one might be 

```
[[0, 0, 1j, 0, 0, 0, 0, 0]
 [0, 1, 0, 0, 0, 0, 0, 0]
 [1j, 0, 0, 0, 0, 0, 0, 0]
 [0, 0, 0, 1, 0, 0, 0, 0]
 [0, 0, 0, 0, 1, 0, 0, 0]
 [0, 0, 0, 0, 0, 1, 0, 0]
 [0, 0, 0, 0, 0, 0, 1, 0]
 [0, 0, 0, 0, 0, 0, 0, 1]]
```

File: cirq_superstaq/custom_gates_test.py [hunk 1]
```diff
@@ -480,6 +480,55 @@ def test_parallel_rgate() -> None:
...
+        np.array(
+            [
+                [0, 1j, 0, 0, 0, 0, 0, 0],
+                [1j, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 0, 0, 1, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1],
```
Comment (richrines1) [inline cirq_superstaq/custom_gates_test.py:522]: why is this targeting the first qubit? i think it should be
```
                [1, 0, 0, 0, 0, 0, 0, 0],
                [0, 1, 0, 0, 0, 0, 0, 0],
                [0, 0, 1, 0, 0, 0, 0, 0],
                [0, 0, 0, 1, 0, 0, 0, 0],
                [0, 0, 0, 0, 1, 0, 0, 0],
                [0, 0, 0, 0, 0, 1, 0, 0],
                [0, 0, 0, 0, 0, 0, 0, 1j],
                [0, 0, 0, 0, 0, 0, 1j, 0],
```

**Comment by vtomole** [comment]
Changed definition of `IITOFFOLI` gate to `cirq.XPowGate(exponent=exponent, global_shift=0.5).controlled(2, [0,0])`.

File: cirq_superstaq/custom_gates_test.py [hunk 2]
```diff
@@ -480,6 +480,30 @@ def test_parallel_rgate() -> None:
     assert circuit.to_qasm(header="", qubit_order=qubits) == expected_qasm
 
 
+def test_icccxpowgate() -> None:
```
Comment (richrines1) [inline cirq_superstaq/custom_gates_test.py:483]:
```suggestion
def test_iitoffoli() -> None:
```

File: cirq_superstaq/custom_gates.py [hunk 3]
```diff
@@ -511,6 +508,12 @@ def _json_dict_(self) -> Dict[str, Any]:
         return cirq.protocols.obj_to_dict_helper(self, ["theta", "phi", "num_copies"])
 
 
+CR = ZX = ZXPowGate()  # standard CR is a full turn of ZX, i.e. exponent = 1
+
+# Inverted iToffoli gate
+IICCX = IITOFFOLI = IICCNOT = cirq.XPowGate(global_shift=0.5).controlled(2, [0, 0])
```
Comment (richrines1) [inline cirq_superstaq/custom_gates.py:514]: imo given how particular the gate is we don't really need to define `IICCNOT` as well, but if we do want to include it it should also be imported in __init__.py

File: cirq_superstaq/custom_gates_test.py [hunk 4]
```diff
@@ -507,96 +507,25 @@ def test_iccxpowgate() -> None:
...
         ),
     )
-
-    print(cirq.unitary(cirq_superstaq.ICCXPowGate().on(*qubits)))
     assert np.allclose(
         cirq.unitary(cirq_superstaq.ICCXPowGate().on(*qubits)),
+        # yapf: disable
         np.array(
             [
-                [
```
Comment (vtomole) [inline cirq_superstaq/custom_gates_test.py]: Link

File: cirq_superstaq/custom_gates_test.py [hunk 5]
```diff
@@ -480,6 +480,55 @@ def test_parallel_rgate() -> None:
...
+                  │
+        1: ───────@────
+                  │
+        2: ───────iX───
+            """
+        ),
+    )
+    assert np.allclose(
+        cirq.unitary(cirq_superstaq.ICCXPowGate().on(*qubits)),
+        # yapf: disable
```
[the remark this answers is not in the document]
Comment (vtomole) [reply cirq_superstaq/custom_gates_test.py:490]: black [mangles](https://github.com/SupertechLabs/cirq-superstaq/pull/207/commits/e9a09d025925adb94ec523ff538f84c9dc73fee0#r847843710) this array. to a point where it's hard to see the the structure of the unitary we are asserting.

[the remark this answers is not in the document]
Comment (richrines1) [reply cirq_superstaq/custom_gates_test.py:490]: ah ok makes sense

File: cirq_superstaq/custom_gates.py (hunk 3 above)
Comment (vtomole) [reply cirq_superstaq/custom_gates.py]: Done.

File: cirq_superstaq/custom_gates_test.py (hunk 2 above)
Comment (vtomole) [reply cirq_superstaq/custom_gates_test.py]: Done.

**Review by richrines1 — APPROVED** [review]
lgtm

**Comment by vtomole** [comment]
Thanks.

**Pull request merged**